sdk/core/azure-core-serializer-json-gson/src/main/java/com/azure/core/serializer/json/gson/models/GsonJsonElementUtils.java

```diff
@@ -0,0 +1,267 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.core.serializer.json.gson.models;
+
+import com.azure.json.JsonReader;
+import com.azure.json.JsonToken;
+import com.azure.json.JsonWriter;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonNull;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Utility class that helps convert between GSON and Azure JSON models.
+ */
+public final class GsonJsonElementUtils {
```

Comment:
Do we want this util to be a public type? Can we move this to implementation package?

Reply:
I was thinking about making this a public type so if external usages of JsonReader wanted to interop with GSON, and for the comment below Jackson, JSON tree representations they could do so without needing to read JSON to GSON's JsonElement then copy it over to the JSON tree models in azure-json.

For now, I will make this package-private, and think on other ways this could be design in the future. Maybe azure-core-serializer-json-gson and azure-core-serializer-json-jackson could have full implementation of all the JSON tree models in azure-json and they just have an overload for wrapping the GSON and Jackson JSON tree models. Ex, GsonJsonNumber(JsonPrimitive) extends JsonNumber.